exampleCourse/questions/demo/randomFakeData/server.py

```diff
@@ -0,0 +1,40 @@
+import datetime
+import random
+
+from faker import Faker
+
+
+def formatDate(date):
+    return date.strftime("%B %d, %Y")
+
+
+def appendEmployee(data, fake, alias, birthdays):
+    name = fake.name()
+    birthday = formatDate(birthdays.pop(0))
+    data["params"][alias] = name
+
+    # Store in array to facilitate rendering list of employees in question.html template
+    data["params"]["employees"].append({"name": name, "birthday": birthday})
+
+
+def generate(data):
+    fake = Faker()
+    # Start with `set` to ensure uniqueness
+    birthdays = set()
+
+    year = 2000
+    while len(birthdays) < 4:
+        birthdays.add(
+            fake.date_between(datetime.date(year, 1, 1), datetime.date(year, 12, 31))
+        )
+
+    # Once sorted, the first birthday belongs to the youngest employee
+    birthdays = sorted(birthdays)
+
+    data["params"]["employees"] = []
+    appendEmployee(data, fake, "youngest", birthdays)
+    appendEmployee(data, fake, "employee1", birthdays)
+    appendEmployee(data, fake, "employee2", birthdays)
+    appendEmployee(data, fake, "employee3", birthdays)
```

Comment:
Totally bikeshedding here, but something about this smells funny - maybe the fact that `appendEmployee` mutates `birthdays`? Consider rewriting like this:

```suggestion
    for i, alias in enumerate(["youngest", "employee1", "employee2", "employee3"]):
        name = fake.name()
        data["params"][alias] = name
        data["params"]["employees"].append({"name": name, "birthday": formatDate(birthdays[i])})
```

Reply:
This could also be done with a `zip` of aliases and birthdays for a cleaner code.
```suggestion
    for alias, birthday in zip(["youngest", "employee1", "employee2", "employee3"], birthdays):
        name = fake.name()
        data["params"][alias] = name
        data["params"]["employees"].append({"name": name, "birthday": formatDate(birthday)})```

Reply:
Totally fair rewrites, I actually thought about doing something like this but was looking at this pretty late 😅